qbraid_qir/qasm3/convert.py

```diff
@@ -16,21 +16,67 @@
 Module containing OpenQASM to QIR conversion functions
 
 """
-from typing import Optional, Union
+from enum import Enum
+from typing import Any, Optional, Union
 
 import openqasm3
 import pyqasm
 from pyqir import Context, Module, qir_module
 
+# from .adaptive_support import QasmQIRAdaptiveVisitor
 from .elements import QasmQIRModule, generate_module_id
 from .exceptions import Qasm3ConversionError
 from .visitor import QasmQIRVisitor
 
 
+class Profile(Enum):
+    """QIR Profile enumeration."""
+
+    BASE = "Base"
+    ADAPTIVE = "AdaptiveExecution"
+
+    @classmethod
+    def from_input(cls, profile: Union["Profile", str]) -> "Profile":
+
+        """Convert string or Profile enum to Profile enum.
+        
+        Args:
+            profile: Either a Profile enum instance or a string representation
+            
+        Returns:
+            Profile enum instance
+            
+        Raises:
+            NotImplementedError: If string doesn't match any valid profile
+            TypeError: If input is neither Profile nor string
+        """
+
+        if isinstance(profile, cls):
+            return profile
+        elif isinstance(profile, str):
+            try:
+                normalized = profile.strip().lower()
+                profile_map = {
+                    "base": cls.BASE,
+                    "adaptive": cls.ADAPTIVE,
+                }
+                return profile_map[normalized]
+            except KeyError:
+                valid = [p.value for p in cls]
+                raise NotImplementedError(f"Invalid profile: {profile}. Valid profiles are: {valid}")
+        else:
+            raise TypeError(f"Profile must be of type Profile or str, not {type(profile).__name__}")
+
+# Export the enum values directly for convenience
+BASE = Profile.BASE
```

Comment:
Let's not export these as `BASE` / `ADAPTIVE` without the `Profile` context don't convey the right meaning